doc/porting_processing.dox

```diff
@@ -0,0 +1,72 @@
+- Derive your algorithms from the new base class QgsProcessingAlgorithm (or a subclass of QgsProcessingAlgorithm), not GeoAlgorithm
+
+- Ensure that your algorithm (or algorithm's parent class) implements the new pure virtual createInstance(self, config) call to return a new instance of the algorithm class. Note that this should not return a copy of the algorithm, but instead a newly constructed instance of the class. If you use a base class in your plugin for all algorithms, you can usually shortcut the createInstance implementation by placing:
+ 
+    def createInstance(self, config={}):
+        return type(self)()
+
+  inside your algorithm base class.
+
+- Input parameters and available outputs must be declared in an implementation of the new pure virtual method initAlgorithm(self, config={})
+
+- The input parameters and outputs classes have been replaced with new c++ versions, which must be used when calling addParameter and addOuput.    
+Map parameters from old classes to new classes
+   - ParameterNumber -> QgsProcessingParameterNumber
+       - constructor arguments in different order
+	   - type specified explicitly, not via min/max/default type
+   - ParameterBoolean -> QgsProcessingParameterBoolean
+   - ParameterCrs -> QgsProcessingParameterCrs
+
+- Use featuresources and sinks wherever possible
+
+Update outputs
+
+- OutputVector -> 
+
+- processAlgorithm has a new signature....
+
+Update retrieval of parameters
+
+- Extent parameters now returned as QgsRectangle. In 2.x extent parameters were returned as a delimited string, leaving the algorithm
+responsible for parsing the string and validating it. In 3.0 this is automatically handled and a QgsRectangle returned, ready for use.
+
+- Crs parameters now returned as QgsCoordinateReferenceSystem objects. In 2.x crs parameters were returned as a string, leaving the algorithm
+responsible for interpreting this string and working out how to convert it to a suitable CRS object. In 3.0 this is automatically handled
+and a QgsCoordinateReferenceSystem object is returned, ready for use. If you require a string version of the CRS, use the
+QgsCoordinateReferenceSystem methods like authid() to obtain a string representation of the CRS in the desired format.
+
+
+Use parameterAsSink instead of getOutputFromName
+
+- Since processing algorithms can now be run in the background, it's important to implement support for cancelation in your
+algorithms. Whenever looping (or before/after calling lengthy operations) listen out for user cancelation via the provided
+feedback object. E.g.
+
+   if feedback.isCanceled():
+       break
+
+- Give feedback via the feedback object, not directly to QgsMessageLog. E.g.
+
+   feedback.pushInfo('Feature {} was missing geometry, skipping'.format(f.id())
+   feedback.reportError('Input layer CRS is not valid, cannot reproject')
+
+- Use geoprocessingexception only for FATAL errors which force a model to terminate. Algorithms should handle common cases
```

Comment:
I suggest to add note about using `QgsProcessingException` instead of `GeoAlgorithmExecutionException`